shop.py

```diff
@@ -0,0 +1,75 @@
+splash = '''
+
+                    ███████╗██╗  ██╗ ██████╗ ██████╗     ██████╗  ██████╗ ████████╗
+                    ██╔════╝██║  ██║██╔═══██╗██╔══██╗    ██╔══██╗██╔═══██╗╚══██╔══╝
+                    ███████╗███████║██║   ██║██████╔╝    ██████╔╝██║   ██║   ██║
+                    ╚════██║██╔══██║██║   ██║██╔═══╝     ██╔══██╗██║   ██║   ██║
+                    ███████║██║  ██║╚██████╔╝██║         ██████╔╝╚██████╔╝   ██║
+                    ╚══════╝╚═╝  ╚═╝ ╚═════╝ ╚═╝         ╚═════╝  ╚═════╝    ╚═╝
+
+                       - = https://github.com/werewolves-devs/werewolf_bot = -
+
+'''
+
+splashes = [
+'It\'s only selling your soul to the devil!',
+'They\'re cheap, honest!'
+]
+
+import discord
+import asyncio
+import random
+import json
+from emoji import emojize, demojize
+
+# Import config data
+import config
+
+client = discord.Client()
+
+async def setup_shop(shop_config):
+    # Sends the shop message and adds reactions
+    embed = discord.Embed(title="Shop (Page 1/1)", description=shop_config["shop_description"], color=0x00ff00)
+    for item in shop_config["items"]:
+        embed.add_field(name="[{}] {}".format(item["emoji"], item["name"]), value="{} {}\n*{}*\n".format(item["price"], shop_config["currency"], item["description"]), inline=False) # Add item to shop
+    message = await client.get_channel(config.shop_channel).send(embed=embed)
+    for item in shop_config["items"]:
+        await message.add_reaction(emojize(item["emoji"], use_aliases=True)) # Add reactions to shop
+    return message # Return the message so we can use it later
+
+async def find_item_from_key(column, query):
+    with open('shop.json') as f:
+        shop_config = json.load(f) # Load item file
+    for item in shop_config["items"]:
+        # print("Testing {} against {}".format(item[column], query)) # This is very useful when trying to find the full emoji name of something
+        if item[column] == query:
+            return item
+
+# Whenever the bot regains his connection with the Discord API.
+@client.event
+async def on_ready():
+    print(' --> Logged in as')
+    print('   | > ' + client.user.name)
+    print('   | > ' + str(client.user.id))
+    # Send shop item message
+    await client.get_channel(config.welcome_channel).send('Shop\'s open for business bois')
+
+    with open('shop.json') as f:
+        shop_config = json.load(f) # Load shop config file
+
+    shop = await setup_shop(shop_config)
+
+    while True:
+        reaction, user = await client.wait_for('reaction_add') # Wait for a user reaction
+        if user != client.user and reaction.message.id == shop.id:
```

Comment:
Besides this spamming the processor, multiple instances of this code are gonna be guaranteed to run simultaneously, effectively forcing players to buy stuff multiple times.
Instead, use **on_reaction_add** or **on_raw_reaction_add**; https://discordpy.readthedocs.io/en/rewrite/api.html#discord.on_reaction_add

Reply:
then how does one access the `shop` variable?

Reply:
It could be stored in a database.
Or, if you want the store to be forgotten when the bot reboots, one doesn't need to store it, as it is already stored in the cache.

Reply:
but `store` is an Object - how do we store it in a database